products/batch_exports/backend/temporal/batch_exports.py

```diff
@@ -761,26 +760,27 @@ async def execute_batch_export_insert_activity(
         initial_interval=dt.timedelta(seconds=initial_retry_interval_seconds),
         maximum_interval=dt.timedelta(seconds=maximum_retry_interval_seconds),
         maximum_attempts=maximum_attempts,
-        non_retryable_error_types=non_retryable_error_types,
     )
 
     try:
-        records_completed = await workflow.execute_activity(
+        result = await workflow.execute_activity(
             activity,
             inputs,
             start_to_close_timeout=start_to_close_timeout,
             heartbeat_timeout=dt.timedelta(seconds=heartbeat_timeout_seconds) if heartbeat_timeout_seconds else None,
             retry_policy=retry_policy,
         )
-        finish_inputs.records_completed = records_completed
+        finish_inputs.records_completed = result.records_completed
+        finish_inputs.bytes_exported = result.bytes_exported
+        if result.error_repr:
+            finish_inputs.latest_error = result.error_repr
+            finish_inputs.status = BatchExportRun.Status.FAILED
 
     except exceptions.ActivityError as e:
         if isinstance(e.cause, exceptions.CancelledError):
             finish_inputs.status = BatchExportRun.Status.CANCELLED
-        elif isinstance(e.cause, exceptions.ApplicationError) and e.cause.type not in non_retryable_error_types:
-            finish_inputs.status = BatchExportRun.Status.FAILED_RETRYABLE
         else:
-            finish_inputs.status = BatchExportRun.Status.FAILED
+            finish_inputs.status = BatchExportRun.Status.FAILED_RETRYABLE
```

Comment:
hmm I'm thinking that maybe `FAILED_RETRYABLE` is not a good name, and `FAILED_INTERNAL` or `FAILED_CRITICAL` would be a better idea.

Reply:
But that would require a migration, so we can update later.

Reply:
yeah, the terminology used here is not too intuitive. I was wondering when looking over the code if it would be better to refer to `internal` and `user` rather than `retryable` and `non-retryable` since it's not obvious, especially to someone not in our team why things are retryable and why they aren't. It feels to me as the distinction should be made between `user` and `internal` errors and then as a consequence of this we don't retry on `user` error, only on `internal` errors. Maybe even this is not great, since there could be some classes of errors that could be considered user errors that we may want to retry a limited number of times (such as connection errors). Perhaps one to think about in future.